okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -158,6 +168,25 @@ public Http2Codec(
         .headers(headersBuilder.build());
   }
 
+  /**
+   * Returns the code of the last ":status" header in {@code headerBlock}, or -1 if there is no
+   * status header in the block. Returning the last status works around a corner case where the
+   * response headers contain both a "100 Continue" status line and its follow-up "200 OK" status
+   * line.
+   */
+  private int statusCode(List<Header> headerBlock) throws IOException {
```

Comment:
Yeah, good call. You can see I went around in circles a bit here. I’ve changed it some more, and addressed my TODO about how to handle multiple blocks of headers in the normal case. It’s a bit ugly (null as sentinel!) but it’s well-hidden.